go/vt/vtctl/workflow/server.go

```diff
@@ -3292,22 +3296,35 @@ func (s *Server) switchWrites(ctx context.Context, req *vtctldatapb.WorkflowSwit
 			return 0, sw.logs(), nil
 		}
 
+		// We stop writes on the source before stopping the source streams so that the
+		// catchup time is lessened and other workflows that we have to migrate such as
+		// intra-keyspace materialize workflows also have a chance to catch up as well
+		// because those are internally generated GTIDs within the keyspace being resharded.
+		// For intra-keyspace materialization streams that we migrate where the source and
+		// target are the keyspace being resharded, we wait for those to catchup in the
+		// stopStreams path before we actually stop them.
+		ts.Logger().Infof("Stopping source writes")
+		if err := sw.stopSourceWrites(ctx); err != nil {
+			sw.cancelMigration(ctx, sm)
+			return handleError(fmt.Sprintf("failed to stop writes in the %s keyspace", ts.SourceKeyspaceName()), err)
+		}
+
 		ts.Logger().Infof("Stopping streams")
-		sourceWorkflows, err = sw.stopStreams(ctx, sm)
+		// Use a shorter context for this since since when doing a Reshard, if there are intra-keyspace
+		// materializations then we have to wait for them to catchup before switching traffic for the
+		// Reshard workflow. We use the timeout where which is used to limit the amount of time that we
```

Comment:
```suggestion
		// Reshard workflow. We use the timeout here which is used to limit the amount of time that we
```

Reply:
I had to guess what you meant here. Feel free to rewrite some other way to make the meaning clear.

Reply:
I made some general improvements to the comment, including fixing the where -> here wording.